subprojects/testing-junit-platform/testing-junit-platform.gradle

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+sourceCompatibility = 1.8
```

Comment:
Not sure this is the correct solution but works for me: move cursor to the error, Alt+Enter and select "set language level to 1.8".

Reply:
Thanks that works. However, this would need to happen automatically when generate the project files with the `idea` task. You'll probably had to set `idea.project.languageLevel = '1.8` (see https://docs.gradle.org/current/dsl/org.gradle.plugins.ide.idea.model.IdeaProject.html#org.gradle.plugins.ide.idea.model.IdeaProject:languageLevel).

Reply:
Thanks! Now I added necessary configuration here: 
https://github.com/gradle/gradle/blob/a4c5eb7b92120e9052a550328bb1dc9d0e5a4b19/gradle/idea.gradle#L6

And it works for me now.

Reply:
Something still seems to be off with the IntelliJ configuration. I am getting the following message upon "Rebuild Project":

```
Information:java: Errors occurred while compiling module 'testingJunitPlatform'
Information:javac 1.8.0_25 was used to compile java sources
Information:2/1/18, 5:34 PM - Compilation completed with 1 error and 148 warnings in 33s 573ms
Error:java: javacTask: source release 1.8 requires target release 1.8
```